server/platform/services/sharedchannel/sync_recv.go

```diff
@@ -79,7 +80,17 @@ func (scs *Service) processSyncMessage(c request.CTX, syncMsg *model.SyncMsg, rc
 		return fmt.Errorf("cannot check channel share state for sync message: %w", err)
 	}
 	if !exists {
-		return fmt.Errorf("cannot process sync message; channel not shared with remote: %w", ErrRemoteIDMismatch)
+		// Update the UI to ensure the link icon is removed
```

Comment:
We need to check not only if the channel and remote have a `SharedChannelRemote`, but if they have one that has been deleted (so the channel was shared before). I'm thinking something like:

```go
if !exists {
    remote, err := scs.server.GetStore().SharedChannel().GetRemoteByIds(targetChannel.Id, rc.RemoteId)
    // check error to exclude sql.ErrNoRows

    if remote && deletedRemote.DeleteAt != 0 {
        return fmt.Errorf("channel is no longer shared: %w", ErrChannelNotShared)
    }

    return fmt.Errorf("cannot process sync message; channel not shared with remote: %w", ErrRemoteIDMismatch)
}
```

or we can directly fetch the remote instead of calling `HasRemote` and check for remote existence, then remote deletion, and have those three branches.

Reply:
"We need to" - just to understand the bar - is this strictly necessary from the core functionality perspective or do we see it as a perf. optimization? We might send a few more  "unshare" message than necessary if the remote doesn't exist at all (never was shared). but I'd argue that that case should be quite rare (fwik messages shouldn't be sent  to remotes that don't have a shared channel relationship) so they should not reach processSyncMessage, so the practical impact should be minimal, unless I miss something.

Reply:
I think you're right, the practical impact right now is going to be minimal. I'd like to distinguish between the errors for correctness shake, but it shouldn't be blocking for the PR to get in.